### DISCUSSION TITLE
Stuck on "Welcome to nginx!"

### DISCUSSION THREAD

**Discussion by camthoms** [mastodon/mastodon#20469] [Q&A] [answered]
I got to the end of the install doc, but now I'm just getting "Welcome to nginx!" instead of the mastodon page. Any help?

**Comment by moniesoke** [comment]
I guess you are installing from source? Once you linked mastodon from site-available to site-enable, and did the certification, you should be able to see mastodon. But certification seem to require that you disable (#) mastodon file's 443 port and enable them and the certificates lines (change example.com to your domain too) after certification.

**Comment by mawann** [comment]
In Mastodon, example.com is not the same as www.example.com.

Try typing this command:
`ls /etc/nginx/sites-enabled/
`
You will see 2 "files" like this (without www):
```
example.com.conf
example.com-ssl.conf

```
Or like this (with www):
```
www.example.com.conf
www.example.com-ssl.conf

```
Let's assume you don't use www. Take a look at the contents of those files.
`cat /etc/nginx/sites-enabled/example.com-ssl.conf
`
You will see something like this:
```
server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;

    server_name example.com;

```
Try restarting nginx.
`systemctl restart nginx.service
`
Then type in the browser like this:
`https://example.com
`

**Reply by moniesoke** [reply]
The install guide will ask you to copy mastodon file from mastodon folder to nginx/site-available. After that you do the "ln" command. Then, in both folders are two files: default and mastodon. The default file will give you welcome to nginx page, and the mastodon will be settings for the mastodon webpage. Maybe we have a different system? I was using ubuntu 22 or 20.

Also I replaced example.com to the domain I bought, so I need to change example.com in mastodon file. Did you buy a domain name and point to your IP in the DNS settings?

I don't know why this happened.

**Reply by camthoms** [reply]
Yeah I only have "mastodon" in the /etc/nginx/sites-enabled/ folder. I deleted the default file

I'm trying to access the site via localhost and the virtual machine's local IP. But now this morning I'm just getting a blank page on both. And now Firefox is giving me a certificate warning

**Reply by moniesoke** [reply]
I got the same thing. But you can access it with app if you have domain name. I can access domain/about for mastodon. I can register new users and sign in and after that I can post stuff. But too many things are not working.

That's why I stopped trying using source code and opt to use the image on digitcoean. I guess there is some problem in its install with source guide. Or maybe there should be more instructions on how to setup the server properly after installing the source code. The original setup is too basic.

**Comment by camthoms** [comment]
I tried accessing the local ipv4 and got a privacy warning: "NET::ERR_CERT_COMMON_NAME_INVALID"

Saying that the security certificate is from my domain

When I continue I get an access denied, http error 403

**Comment by camthoms** [comment, marked as answer]
I'm a doofus, forgot to forward port 443 only forwarded port 80
